doc/sphinx/proofs/writing-proofs/rewriting.rst

```diff
@@ -926,14 +909,257 @@ the conversion in hypotheses :n:`{+ @ident}`.
+   may be used to make the change in the selected hypotheses and/or the conclusion.
+   The `at` clause of :n:`@goal_occurrences` is not supported.
+
+.. tacn:: transitivity @one_term
+
+   Changes a goal that has the form :g:`t=u`
+   into the two subgoals :n:`t=@one_term` and :n:`@one_term=u`.
+
+   .. tacn:: etransitivity
+
+      This tactic behaves like :tacn:`transitivity`, using a fresh evar instead of
+      a concrete :token:`one_term`.
+
+Equality and inductive sets
+---------------------------
+
+We describe in this section some special purpose tactics dealing with
+equality and inductive sets or types. These tactics use the
+equality `eq:forall (A:Type), A -> A -> Prop`, simply written with the infix
+symbol `=`.
+
+.. tacn:: decide equality
+
+   Solves a goal of the form :g:`forall x y : R, {x = y} + {~ x = y}`,
+   where :g:`R` is an inductive type such that its constructors do not take
+   proofs or functions as arguments, nor objects in dependent types. It
+   solves goals of the form :g:`{x = y} + {~ x = y}` as well.
+
+.. tacn:: compare @one_term__1 @one_term__2
+
+   Compares two :n:`@one_term`\s of an
+   inductive datatype. If :g:`G` is the current goal, it leaves the
+   sub-goals :n:`@one_term__1 = @one_term__2 -> G` and :n:`~ @one_term__1 = @one_term__2 -> G`.
+   The type of the :n:`@one_term`\s must satisfy the same restrictions as in the
+   tactic ``decide equality``.
+
+.. tacn:: simplify_eq {? @destruction_arg }
+
+   Let :n:`@term` be the proof of a statement with the conclusion :n:`@term__1 = @term__2`.
+   If :n:`@term__1` and :n:`@term__2` are structurally different (in the sense
+   described for the tactic :tacn:`discriminate`), then the tactic
+   ``simplify_eq`` behaves as :n:`discriminate @term`, otherwise it behaves as
+   :n:`injection @term`.
+
+   If the current goal has form :g:`t1 <> t2`, it behaves as
+   :n:`intro @ident; simplify_eq @ident`.
+
+   :n:`simplify_eq @natural` is equivalent to :n:`intros until @natural` then
+   :n:`simplify_eq @ident` where :n:`@ident` is the identifier for the last
+   introduced hypothesis.
+
+   If :n:`@bindings` are provided in :n:`@destruction_arg`, they are used to
+   instantiate parameters or hypotheses of :n:`@term`.
+
+.. note::
+   If some quantified hypothesis of the goal is named :n:`@ident`,
+   then :n:`simplify_eq @ident` first introduces the hypothesis in the local
+   context using :n:`intros until @ident`.
+
+.. tacn:: esimplify_eq {? @destruction_arg }
+
+   This works the same as :tacn:`simplify_eq` but if the type of :n:`@term` or the
+   type of the hypothesis referred to by :n:`@natural` has uninstantiated
+   parameters, these parameters are left as existential variables.
 
-   The form :n:`@tactic in {+, @ident }` applies :token:`tactic` (any of the
-   conversion tactics listed in this section) to the hypotheses :n:`{+ @ident}`.
+.. tacn:: dependent rewrite {? {| -> | <- } } @one_term {? in @ident }
```

Comment:
It may be worth mentioning `inversion_sigma` from the standard library here?  It splits all hypotheses of the form `existT _ _ _ = existT _ _ _` into equalities that can be used directly by `rewrite` or `subst`, and is not subject to the same errors as `dependent rewrite` such as "Cannot find a well-typed generalization of the goal that makes the proof progress." on
```coq
Goal existT (fun T => T) nat 1 = existT (fun T => T) nat 2 -> 1 = 2.
  intro H.
  dependent rewrite H.
```